os/os-mac.h

```diff
@@ -117,3 +123,58 @@ static inline bool os_cpu_has(cpu_features feature)
 }
 
 #endif
+
+/*
+ * discard_pages should not be run within Rosetta. Native arm64 or x86 only. 
```

Comment:
Is there a way of checking if this is happening? What happens if you DO do the `msync(MS_INVALIDATE)` under Rosetta? :-)